front, editoast: fix missing fields in scenario and study search results

Synar opened the pull request:
Close [#7095](https://github.com/OpenRailAssociation/osrd/issues/7095)

Paced trains count was used by the front in addition to trains count when using the search endpoint, and was present in ScenarioResponse but not SearchResultItemScenario.

Study type, business code and study code were used by the front when using the search endpoint, and were present in Study but not SearchResultItemStudy.

One thing of note is that there are many fields (ids, counts, ...) typed as u64 in search.rs but i64 in other models and views. Is there a reason for this? If not, it's probably not super important, but do you now which we should prefer?

Synar committed:
editoast: fix missing fields in scenario and study search results

Paced trains count was used by the front in addition to trains count when using the search endpoint,
and was present in ScenarioResponse but not SearchResultItemScenario.

Study type, business code and study code were used by the front when using the search endpoint,
and were present in Study but not SearchResultItemStudy.

Signed-off-by: Alice Khoudli <alice.khoudli@polytechnique.org>

leovalais reviewed (approved):
Thanks for the fix!


> One thing of note is that there are many fields (ids, counts, ...) typed as u64 in search.rs but i64 in other models and views. Is there a reason for this? If not, it's probably not super important, but do you now which we should prefer?

Ideally we'd use `u64` everywhere, but diesel types as `i64` because SQL doesn't have unsigned integers. They have always leaked to the whole codebase, ideally they'd be converted as early as possible, but it's kinda late for that now.

Synar committed:
front: fix scenario and study card prop types

Scenario and study search results were cast into improper types
(that did not correspond to their actual openapi types)
in order to be passed as props to scenario and study cards.

This caused at least one bug, where some fields were used by the cards
despite not being present in the search results.

Thus change the scenario and study card prop types to correspond to the actual types
they can receive from api requests.

Signed-off-by: Alice Khoudli <alice.khoudli@polytechnique.org>

Synar commented:
Thanks for the explanation!

kmer2016 reviewed (approved):
LGTM
